diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/UselessSupertype.kt

```diff
@@ -0,0 +1,139 @@
+import com.pinterest.ktlint.core.ast.ElementType.SUPER_TYPE_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.SUPER_TYPE_LIST
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_REFERENCE
+import com.pinterest.ktlint.core.ast.parent
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.USELESS_SUPERTYPE
+import org.cqfn.diktat.ruleset.utils.*
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.psiUtil.siblings
+
+/**
+ * rule 6.1.5
+ * Explicit supertype qualification should not be used if there is not clash between called methods
+ * fixme can't fix supertypes that are defined in other files.
+ */
+class UselessSupertype(private val configRules: List<RulesConfig>) : Rule("useless-override") {
+
+    companion object {
+        private val SUPER_TYPE = listOf(SUPER_TYPE_CALL_ENTRY, SUPER_TYPE_ENTRY)
+    }
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == CLASS)
+            checkClass(node)
+    }
+
+    private fun checkClass(node: ASTNode) {
+        val superNodes = node.findChildByType(SUPER_TYPE_LIST)?.findAllNodesWithCondition({ it.elementType in SUPER_TYPE })?.takeIf { it.isNotEmpty() } ?: return
+        val qualifiedSuperCalls = node.findAllNodesWithSpecificType(DOT_QUALIFIED_EXPRESSION)
+                .mapNotNull { findFunWithSuper(it) }.ifEmpty { return }
+        if (superNodes.size == 1) {
+            qualifiedSuperCalls.map { removeSupertype(it.first) }
+        } else {
+            handleManyImpl(superNodes, qualifiedSuperCalls)
+        }
+    }
+
+    private fun handleManyImpl(superNodes: List<ASTNode>, overrideNodes: List<Pair<ASTNode, ASTNode>>) {
+        val uselessSuperType = findAllSupers(superNodes, overrideNodes.map { it.second.text })?.filter { it.value == 1 }?.map { it.key }
+                ?: return
+        overrideNodes
+                .filter {
+                    it.second.text in uselessSuperType
+                }.map {
+                    removeSupertype(it.first)
+                }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun removeSupertype(node: ASTNode) {
+        USELESS_SUPERTYPE.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset, node) {
+            val startNode = node.parent({ it.elementType == SUPER_EXPRESSION })!!.findChildByType(REFERENCE_EXPRESSION)!!
+            val lastNode = startNode.siblings(true).last()
+            startNode.treeParent.removeRange(startNode.treeNext, lastNode)
+            startNode.treeParent.removeChild(lastNode)
```

Comment:
So, here you remove all children of `SUPER_EXPRESSION`. But the `SUPER_EXPRESSION` node itself is left in the tree? You should remove it explicitly too, even though it's empty.

Reply:
No, I shouldn't remove `SUPER_EXPRESSION`.